lib/RestructureCFG/InlineDispatcherSwitch.cpp

```diff
@@ -737,13 +737,140 @@ inlineDispatcherSwitchImpl(ASTTree &AST,
   rc_return Node;
 }
 
-ASTNode *inlineDispatcherSwitch(ASTTree &AST, ASTNode *RootNode) {
+static RecursiveCoroutine<ASTNode *>
+simplifySuperfluousNonLocalCFImpl(ASTNode *Node,
+                                  const FallThroughScopeTypeMap
+                                    &FallThroughScopeMap) {
+  switch (Node->getKind()) {
+  case ASTNode::NK_List: {
+    SequenceNode *Seq = llvm::cast<SequenceNode>(Node);
+
+    // In place of a sequence node, we just need to inspect all the nodes in the
+    // sequence
+    bool ShouldSimplify = false;
+    for (ASTNode *&N : Seq->nodes()) {
+      if (not ShouldSimplify) {
+
+        // Compute the `FallThrough information after each `SequencenNode` is
+        // processed
+        N = rc_recur simplifySuperfluousNonLocalCFImpl(N, FallThroughScopeMap);
+        FallThroughScopeType ReturnType = FallThroughScopeType::Return;
+        ShouldSimplify = (FallThroughScopeMap.at(N) == ReturnType);
+      } else {
+        N = nullptr;
+      }
+    }
+
+    // In this beautify, it may be that a dispatcher switch is completely
+    // removed, therefore leaving a `nullptr` in a `SequenceNode`. We remove all
+    // these `nullptr` from the `SequenceNode` after the processing has taken
+    // place
+    Seq->removeNode(nullptr);
```

Comment:
Is this guaranteed to always remove all of the `nullptr`s left? Or only the first one?

Reply:
Yes, the `removeNode` method of  `SequenceNode` takes care of removing all the instances of empty nodes